pkg/kubelet/kubelet_test.go

```diff
@@ -2793,68 +2817,109 @@ func TestHandlePodResourcesResize(t *testing.T) {
 	}
 
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			oldGOOS := goos
-			defer func() { goos = oldGOOS }()
-			if tt.goos != "" {
-				goos = tt.goos
-			}
-			kubelet.statusManager = status.NewFakeManager()
+		for _, isSidecarContainer := range []bool{false, true} {
+			t.Run(tt.name, func(t *testing.T) {
+				oldGOOS := goos
+				defer func() { goos = oldGOOS }()
+				if tt.goos != "" {
+					goos = tt.goos
+				}
+				kubelet.statusManager = status.NewFakeManager()
+
+				var originalPod *v1.Pod
+				if isSidecarContainer {
+					originalPod = testPod2.DeepCopy()
+					originalPod.Spec.InitContainers[0].Resources.Requests = tt.originalRequests
+				} else {
```

Comment:
```go
var originalPod *v1.Pod
var originalCtr *v1.Container
if isSidecarContainer {
	originalPod = testPod2.DeepCopy()
	originalCtr = &originalPod.Spec.InitContainers[0]
} else {
	originalPod = testPod1.DeepCopy()
	originalCtr = &originalPod.Spec.Containers[0]
}
originalCtr.Resources.Requests = tt.originalRequests
```

Then you can use `originalCtr` below instead of branching on `isSidecarContainer`